cmd/talosctl/cmd/mgmt/cluster/create.go

```diff
@@ -128,12 +158,14 @@ var createCmd = &cobra.Command{
 	Long:  ``,
 	Args:  cobra.NoArgs,
 	RunE: func(cmd *cobra.Command, args []string) error {
-		return cli.WithContext(context.Background(), create)
+		return create(cmd.Flags())
 	},
 }
 
 //nolint:gocyclo,cyclop
-func create(ctx context.Context) (err error) {
+func create(flags *pflag.FlagSet) (err error) {
+	ctx := context.Background()
+
 	if masters < 1 {
 		return fmt.Errorf("number of masters can't be less than 1")
 	}
```

Comment:
This introduces big but subtle change - `cli.WithContext` contains a logic which will cancel passed context if there was `SIGINT` or `SIGTERM` signal. Your codes doesn't do that, so it will no longer properly shutdown in case of signal.

I suggest you change this code to:
```go
	RunE: func(cmd *cobra.Command, args []string) error {
		return cli.WithContext(context.Background(), func(ctx context.Context) error {
			create(ctx, cmd.Flags())
		})
	},
}

//nolint:gocyclo,cyclop
func create(ctx context.Context, flags *pflag.FlagSet) (err error) {
	if masters < 1 {
		return fmt.Errorf("number of masters can't be less than 1")
	}
```

Reply:
I didn't know that! thanks